tensorflow_datasets/structured/movielens.py

```diff
@@ -0,0 +1,422 @@
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""MovieLens dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import csv
+import os
+import textwrap
+import codecs
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """
+@article{10.1145/2827872,
+author = {Harper, F. Maxwell and Konstan, Joseph A.},
+title = {The MovieLens Datasets: History and Context},
+year = {2015},
+issue_date = {January 2016},
+publisher = {Association for Computing Machinery},
+address = {New York, NY, USA},
+volume = {5},
+number = {4},
+issn = {2160-6455},
+url = {https://doi.org/10.1145/2827872},
+doi = {10.1145/2827872},
+journal = {ACM Trans. Interact. Intell. Syst.},
+month = dec,
+articleno = {19},
+numpages = {19},
+keywords = {Datasets, recommendations, ratings, MovieLens}
+}
+"""
+
+_DESCRIPTION = """
+This dataset describes 5-star rating from MovieLens,
+a movie recommendation service.
+Users were selected at random for inclusion.
+There are 5 versions included: 25m, latest-small, 100k, 1m, 20m.
+In all datasets, the movies data and ratings data are joined on "movieId".
+The 25m dataset, latest-small dataset, and 1m dataset have the same schema
+and they do not include demographic data.
+The 25m dataset is the latest stable version of the MovieLens dataset.
+It is recommended for research purposes.
+The latest-small dataset is a small dataset with
+the same schema as 25m and 20m.
+It is updated over time by GroupLens and its checksum might need updating.
```

Comment:
Is this information that the end user needs to know?

Reply:
+1 Consider adding a section called Attribute Information which will have field descriptions and that seems more relevant to dataset user.
See example in: https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/structured/forest_fires.py

Reply:
> Is this information that the end user needs to know?

Regarding "the latest-small dataset is a small dataset with the same schema as 25m and 20m," I think if would be helpful if the user wants to train a model on one dataset and use it on another dataset. From this sentence the user would know a model trained on one of latest-small, 25m, or 20m can be used on the other two datasets.

Regarding the checksum update part, considering there is currently no mechanism that can automatically update the checksum, the user might run into a checksum mismatch issue in the future. I think including this heads up would help users and the TFDS team find the source of the problem when they run into a checksum mismatch.



> +1 Consider adding a section called Attribute Information which will have field descriptions and that seems more relevant to dataset user.
> See example in: https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/structured/forest_fires.py

I have added a list of feature descriptions. Please see 333dc4fec316388b6440cb7baf64b755905c724e.